src/components/SignUp.js

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useContext } from "react";
+import { UserContext } from "../contexts/UserContext";
+import FormBox from "../styled/FormBox";
+import Nav from "./Nav";
+
+const SignUp = () => {
+  const { addNewUser } = useContext(UserContext);
+  // Local New User State Construction
+  const [newUser, setNewUser] = useState({
+    username: "",
+    password: "",
+    repassword: "",
+  });
+
+  // Input Field Handling
+  const handleChange = (e) => {
+    setNewUser({ ...newUser, [e.target.name]: e.target.value });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    addNewUser(newUser);
```

Comment:
What happens if there is an error when trying to add a new user?

Reply:
@travis yea. I have yet to actually hookup the user addition side of things. These files in the PR a really just the current state of things but most of the work on this I've done for the pagination.